### PR TITLE
Add recipe for librealsense

### PR DESCRIPTION
**Pull request by Gnimuc**

### CODE REVIEW

File: L/librealsense/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_GRAPHICAL_EXAMPLES=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_UNIT_TESTS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_WITH_OPENMP=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DENFORCE_METADATA=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_PYTHON_BINDINGS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_MATLAB_BINDINGS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_OPENNI2_BINDINGS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_NODEJS_BINDINGS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DBUILD_CSHARP_BINDINGS=false"
+CMAKE_FLAGS="${CMAKE_FLAGS} -DENABLE_CCACHE=false"
```
Comment (giordano) [inline L/librealsense/build_tarballs.jl]: Any particular reason for disabling ccache?  It makes rebuilds much faster

File: L/librealsense/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+cmake ${CMAKE_FLAGS} ..
+make -j${nproc}
+make install
+
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(supported_platforms()) do p
+    p isa Linux && p.libc == :glibc return true
```
Comment (giordano) [inline L/librealsense/build_tarballs.jl]: Plain `if`...`else if` is better here.

File: L/librealsense/build_tarballs.jl (hunk 1 above)
Comment (Gnimuc) [reply L/librealsense/build_tarballs.jl]: I didn't set `BINARYBUILDER_USE_CCACHE` on my machine. I thought this CMake flag would take no effect, but yes, it should be turned on when building on Azure.

File: L/librealsense/build_tarballs.jl (hunk 2 above)
Comment (Gnimuc) [reply L/librealsense/build_tarballs.jl]: It looks like only the following 5 cross-compile environments are officially supported by Intel, so maybe it's better to directly list these platforms.
```
Linux(:i686, libc=:glibc)                    
Linux(:x86_64, libc=:glibc)                  
Linux(:aarch64, libc=:glibc)                 
Linux(:armv7l, libc=:glibc, call_abi=:eabihf)
MacOS(:x86_64)
```

**Comment by Gnimuc** [comment]
The failure on macOS looks like an upstream issue, I'm going to check it again locally.

**Comment by giordano** [comment]
If I'm reading

https://github.com/IntelRealSense/librealsense/blob/3b01c085a5380bebea846ab342ecca89d659ddf3/CMake/unix_config.cmake#L9

correctly, this won't work when we'll add support for `armv6l` :man_facepalming:

**Comment by Gnimuc** [comment]
The error on macOS is caused by an undeclared identifier 'EVFILT_EXCEPT':
```
se-file.dir/rosbag/rostime/src/time.cpp.o -c /workspace/srcdir/librealsense/third-party/realsense-file/rosbag/rostime/src/time.cpp
[22:56:03] /workspace/srcdir/librealsense/third-party/libtm/libtm/src/infra/Poller_bsd.cpp:35:37: error: use of undeclared identifier 'EVFILT_EXCEPT'
[22:56:03]     const int Poller::EXCEPT_MASK = EVFILT_EXCEPT;
[22:56:03]                                     ^
[22:56:03] 1 error generated.
[22:56:03] make[2]: *** [third-party/libtm/libtm/src/CMakeFiles/tm.dir/build.make:232: third-party/libtm/libtm/src/CMakeFiles/tm.dir/infra/Poller_bsd.cpp.o] Error 1
[22:56:03] make[2]: Leaving directory '/workspace/srcdir/librealsense/build'
[22:56:03] make[1]: *** [CMakeFiles/Makefile2:332: third-party/libtm/libtm/src/CMakeFiles/tm.dir/all] Error 2
[22:56:03] make[1]: *** Waiting for unfinished jobs....
[22:56:13] [ 31%] Linking CXX static library librealsense-file.a
```

I tried to build libtm on my mac and there is no error. After some googling, I find `EVFILT_EXCEPT` is valid on macOS 10.13+, but not 10.11. What's the macOS version used in BB2?

**Comment by giordano** [comment]
> I tried to build libtm on my mac and there is no error. After some googling, I find EVFILT_EXCEPT is valid on macOS 10.13+, but not 10.11. What's the macOS version used in BB2?

We're using 10.10 by default.

**Comment by Gnimuc** [comment]
According to [this response](https://github.com/IntelRealSense/librealsense/issues/5567#issuecomment-570045606), libtm will be totally removed in the next release, so this won't be a problem in long short. is it good to merge this dirty-patched version?

**Comment by giordano** [comment]
> is it good to merge this dirty-patched version?

If it's good for you, we can go.  We have lots of dirty patches

**Comment by Gnimuc** [comment]
Let's merge this, so I can start upgrading the wrapper package to use artifacts.

**Pull request merged**